### ISSUE TITLE
Stops working deoplete when fails to gather results in _get_tagfiles function.

### ISSUE THREAD

**Issue by carlitux** [Shougo/neoinclude.vim#15]
``` bash
[deoplete] Traceback (most recent call last):
[deoplete]   File "/Users/carlitux/.cache/dein/repos/github.com/Shougo/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 47, in completion_begin
[deoplete]     complete_position, candidates = self.gather_candidates(context)
[deoplete]   File "/Users/carlitux/.cache/dein/repos/github.com/Shougo/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 81, in gather_candidates
[deoplete]     results = self.gather_results(context)
[deoplete]   File "/Users/carlitux/.cache/dein/repos/github.com/Shougo/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 132, in gather_results
[deoplete]     context['candidates'] = source.gather_candidates(context)
[deoplete]   File "/Users/carlitux/.cache/dein/repos/github.com/Shougo/deoplete.nvim/rplugin/python3/deoplete/sources/tag.py", line 32, in gather_candidates
[deoplete]     self.__make_cache(context)
[deoplete]   File "/Users/carlitux/.cache/dein/repos/github.com/Shougo/deoplete.nvim/rplugin/python3/deoplete/sources/tag.py", line 43, in __make_cache
[deoplete]     for filename in self.__get_tagfiles():
[deoplete]   File "/Users/carlitux/.cache/dein/repos/github.com/Shougo/deoplete.nvim/rplugin/python3/deoplete/sources/tag.py", line 55, in __get_tagfiles
[deoplete]     'exists', '*neoinclude#include#get_tag_files') else []
[deoplete]   File "/usr/local/lib/python3.5/site-packages/neovim/api/nvim.py", line 213, in call
[deoplete]     return self.request('vim_call_function', name, args, **kwargs)
[deoplete]   File "/usr/local/lib/python3.5/site-packages/neovim/api/nvim.py", line 129, in request
[deoplete]     res = self._session.request(name, *args, **kwargs)
[deoplete]   File "/usr/local/lib/python3.5/site-packages/neovim/msgpack_rpc/session.py", line 98, in request
[deoplete]     raise self.error_wrapper(err)
[deoplete] neovim.api.nvim.NvimError: b'Vim(return):E121: Undefined variable: b:node_root'
[deoplete] An error has occurred. Please execute :messages command.
```

**Comment by Shougo** [comment]
I cannot reproduce the error.
Please create minimal init.vim and reproduce ways from Vim starting.

**Comment by Shougo** [comment]
Your neovim and neovim-python and neoinclude are the latest version?

**Comment by Shougo** [comment]
Please use this.

https://github.com/tweekmonster/nvim-python-doctor

**Comment by carlitux** [comment]
Follow this:
1. nvim test.js
2. type: require('path');
3. see messages and you will look the errors

**Comment by Shougo** [comment]
> Please create minimal init.vim and reproduce ways from Vim starting.

minimal init.vim is missing.

**Comment by zolotyh** [comment]
I have the same problem with js project.

When I call :NeoIncludeMakeCache

E121: Undefined variable: b:node_root
E15: Invalid expression: b:node_root . "/node_modules/" . a:name
E121: Undefined variable: b:node_root
E15: Invalid expression: b:node_root . "/node_modules/" . a:name

**Comment by zolotyh** [comment]
I think, the problem is with https://github.com/moll/vim-node

**Comment by carlitux** [comment]
@Shougo I was testing and I reproduced with this file test.js

nvim -u minimal.vim test.js

``` js
var test = require('test');
tes
```

minimal.vim

``` vim
set runtimepath+=./deoplete.nvim/
set runtimepath+=./neoinclude.vim/
set runtimepath+=./deoplete-ternjs/
set runtimepath+=./vim-node/
let g:deoplete#enable_at_startup = 1
```

I saw that when we add vim-node as @zolotyh told nvim start raising the error. Looks like deoplate start gathering the candidates when deoplete-ternjs is installed then try to autocomplete and I don't know where neoinclude enters here to raise the error but only happens when **require** keyword is on file.

Also I tried with:

``` js
import test from 'path';
test
```

And all is working.
